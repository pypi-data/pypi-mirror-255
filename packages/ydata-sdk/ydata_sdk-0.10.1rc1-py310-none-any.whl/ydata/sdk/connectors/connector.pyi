from pathlib import Path
from typing import Dict, Optional, Union
from ydata.sdk.common.client import Client
from ydata.sdk.common.types import Project, UID
from ydata.sdk.connectors._models.connector_list import ConnectorsList
from ydata.sdk.connectors._models.connector_type import ConnectorType
from ydata.sdk.connectors._models.credentials.credentials import Credentials
from ydata.sdk.utils.model_mixin import ModelFactoryMixin

class Connector(ModelFactoryMixin):
    def __init__(self, connector_type: Union[ConnectorType, str] = ..., credentials: Optional[Dict] = ..., name: Optional[str] = ..., project: Optional[Project] = ..., client: Optional[Client] = ...) -> None: ...
    @property
    def uid(self) -> UID: ...
    @property
    def type(self) -> ConnectorType: ...
    @property
    def project(self) -> Project: ...
    @staticmethod
    def get(uid: UID, project: Optional[Project] = ..., client: Optional[Client] = ...) -> Connector: ...
    @staticmethod
    def create(connector_type: Union[ConnectorType, str], credentials: Union[str, Path, Dict, Credentials], name: Optional[str] = ..., project: Optional[Project] = ..., client: Optional[Client] = ...) -> Connector: ...
    @staticmethod
    def list(project: Optional[Project] = ..., client: Optional[Client] = ...) -> ConnectorsList: ...
