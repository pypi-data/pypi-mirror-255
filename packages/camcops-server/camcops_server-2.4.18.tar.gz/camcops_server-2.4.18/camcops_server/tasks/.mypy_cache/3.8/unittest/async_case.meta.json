{"data_mtime": 1658152101, "dep_lines": [1, 2, 4, 1, 1, 1], "dep_prios": [5, 5, 5, 5, 20, 20], "dependencies": ["collections.abc", "typing_extensions", "unittest.case", "builtins", "pyexpat.model", "pyexpat.errors"], "hash": "a2b98e68ebc998006856f80b63f65acbe845d8246ad4315106090373c0d7e587", "id": "unittest.async_case", "ignore_all": true, "interface_hash": "1a2cf0fa859ea5e75ec0d5afbc1f6d9f7548bab5ac1372174b07ef5e9a1b9ead", "mtime": 1658152086, "options": {"allow_redefinition": false, "allow_untyped_globals": false, "always_false": [], "always_true": [], "bazel": false, "check_untyped_defs": false, "disallow_any_decorated": false, "disallow_any_explicit": false, "disallow_any_expr": false, "disallow_any_generics": false, "disallow_any_unimported": false, "disallow_incomplete_defs": false, "disallow_subclassing_any": false, "disallow_untyped_calls": false, "disallow_untyped_decorators": false, "disallow_untyped_defs": false, "follow_imports": "normal", "follow_imports_for_stubs": false, "ignore_errors": false, "ignore_missing_imports": false, "implicit_reexport": true, "local_partial_types": false, "mypyc": false, "no_implicit_optional": false, "platform": "linux", "plugins": [], "show_none_errors": true, "strict_concatenate": false, "strict_equality": false, "strict_optional": true, "strict_optional_whitelist": null, "warn_no_return": true, "warn_return_any": false, "warn_unreachable": false, "warn_unused_ignores": false}, "path": "/home/martinb/.virtualenvs/camcops/lib/python3.8/site-packages/mypy/typeshed/stdlib/unittest/async_case.pyi", "plugin_data": null, "size": 392, "suppressed": [], "version_id": "0.961"}