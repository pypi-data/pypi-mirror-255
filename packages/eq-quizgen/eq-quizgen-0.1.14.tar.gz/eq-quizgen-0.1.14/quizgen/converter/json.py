"""
Convert a quiz to JSON.
"""

class JSONConverter(object):
    def __init__(self, **kwargs):
        super().__init__()

    def convert_quiz(self, quiz, **kwargs):
        return quiz.to_json()
