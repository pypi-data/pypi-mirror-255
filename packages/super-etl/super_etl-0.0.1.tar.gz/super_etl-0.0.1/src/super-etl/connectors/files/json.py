# -*- coding: UTF-8 -*-
#######################################
# JSON File
#######################################
#
# Principle: https://en.wikipedia.org/wiki/JSON
# Library: pandas (https://github.com/pandas-dev/pandas)
# Author: Rik Heijmann
#

from logger import log
import modin.pandas as pandas


####################
# Extract
####################
def get_data(file):
    return pandas.read_json(file)

# TODO: Does currently not support extracting nested fields.
