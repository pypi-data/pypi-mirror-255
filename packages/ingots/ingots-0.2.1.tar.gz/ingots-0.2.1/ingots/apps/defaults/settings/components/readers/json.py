__all__ = ("JsonConfigsFileReader",)

import json
import logging
import typing as t

from .base import BaseConfigsTextFileReader

if t.TYPE_CHECKING:
    ...


logger = logging.getLogger(__name__)


class JsonConfigsFileReader(BaseConfigsTextFileReader):
    supported_extensions = ["json"]

    def parse_content(self, content: str) -> dict[str, t.Any]:
        configs = json.loads(content)
        return configs
