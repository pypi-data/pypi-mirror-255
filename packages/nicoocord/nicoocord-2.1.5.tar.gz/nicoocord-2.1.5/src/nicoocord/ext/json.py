import json
from typing import Union


class JsonParser:
    """
    A simple JSON parser class.

    Private methods
    ---------------
        `__read__`` -> Reads the file and returns a dict.

    Public methods
    --------------
    ``write`` -> Writes data to the file.
        
    ``get`` -> Gets a value from the file.
     """
    def __init__(self, file: str) -> None:
        self.file = file
        self.loaded = self.__read__()

    def __read__(self) -> dict:
        with open(self.file, "r") as f:
            return json.load(f)
        
    def write(self, data: dict) -> None:
        with open(self.file, "w") as f:
            json.dump(data, f, indent=4, ensure_ascii=False)

    def get(self, key: str) -> Union[str, int, float, bool, list, dict]:
        try:
            return self.loaded[key]
        except KeyError:
            raise KeyError(f"Key {key} not found in {self.file}")
