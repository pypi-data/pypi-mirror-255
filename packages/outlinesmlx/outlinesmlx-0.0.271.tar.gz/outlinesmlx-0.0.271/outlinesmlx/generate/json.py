"""
Copyright 2023- The Outlines developers

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
"""


import json as pyjson
from functools import singledispatch
from typing import Callable, Optional, Union

from pydantic import BaseModel

from outlines.fsm.json_schema import build_regex_from_object, get_schema_from_signature
from outlinesmlx.samplers import Sampler, multinomial

from .regex import regex


@singledispatch
def json(
    model,
    schema_object: Union[str, object, Callable],
    max_tokens: Optional[int] = None,
    sampler: Sampler = multinomial(),
):
    if isinstance(schema_object, type(BaseModel)):
        schema = pyjson.dumps(schema_object.model_json_schema())
        regex_str = build_regex_from_object(schema)
        generator = regex(model, regex_str, max_tokens, sampler)
        generator.format_sequence = lambda x: schema_object.parse_raw(x)
    elif callable(schema_object):
        schema = pyjson.dumps(get_schema_from_signature(schema_object))
        regex_str = build_regex_from_object(schema)
        generator = regex(model, regex_str, max_tokens, sampler)
        generator.format_sequence = lambda x: pyjson.loads(x)
    elif isinstance(schema_object, str):
        schema = schema_object
        regex_str = build_regex_from_object(schema)
        generator = regex(model, regex_str, max_tokens, sampler)
        generator.format_sequence = lambda x: pyjson.loads(x)
    else:
        raise ValueError(
            f"Cannot parse schema {schema_object}. The schema must be either "
            + "a Pydantic object, a function or a string that contains the JSON "
            + "Schema specification"
        )

    return generator

