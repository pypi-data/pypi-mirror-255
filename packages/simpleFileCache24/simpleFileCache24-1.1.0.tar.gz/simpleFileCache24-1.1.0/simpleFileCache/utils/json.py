try:
    import orjson

    def read_json(filename :str):
        with open(filename, 'rb') as f:
            return orjson.loads(f.read())

    def save_json(filename :str, data :str):
        with open(filename, 'wb') as f:
            f.write(orjson.dumps(data))    
    
except ImportError:

    import json

    def read_json(filename :str):
        with open(filename, 'r') as f:
            return json.load(f)

    def save_json(filename :str, data :str):
        with open(filename, 'w') as f:
            json.dump(data, f)