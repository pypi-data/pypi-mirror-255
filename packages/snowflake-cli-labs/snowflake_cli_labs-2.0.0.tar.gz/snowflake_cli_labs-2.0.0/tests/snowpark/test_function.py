import json
from pathlib import Path
from textwrap import dedent
from unittest import mock

from snowflake.connector import ProgrammingError


@mock.patch("snowflake.connector.connect")
@mock.patch("snowflake.cli.plugins.snowpark.commands.ObjectManager")
def test_deploy_function(
    mock_object_manager,
    mock_connector,
    mock_ctx,
    runner,
    project_directory,
):
    mock_object_manager.return_value.describe.side_effect = ProgrammingError(
        "does not exist or not authorized"
    )
    ctx = mock_ctx()
    mock_connector.return_value = ctx
    with project_directory("snowpark_functions") as project_dir:
        result = runner.invoke(
            [
                "snowpark",
                "deploy",
            ],
            catch_exceptions=False,
        )

    assert result.exit_code == 0, result.output
    assert ctx.get_queries() == [
        "create stage if not exists dev_deployment comment='deployments managed by snowcli'",
        f"put file://{Path(project_dir).resolve()}/app.zip @dev_deployment/my_snowpark_project"
        f" auto_compress=false parallel=4 overwrite=True",
        dedent(
            """\
            create or replace function func1(a string default 'default value', b variant)
            returns string
            language python
            runtime_version=3.10
            imports=('@dev_deployment/my_snowpark_project/app.zip')
            handler='app.func1_handler'
            packages=()
            """
        ).strip(),
    ]


@mock.patch("snowflake.connector.connect")
@mock.patch("snowflake.cli.plugins.snowpark.commands.ObjectManager")
def test_deploy_function_with_external_access(
    mock_object_manager,
    mock_connector,
    mock_ctx,
    runner,
    project_directory,
):
    mock_object_manager.return_value.show.return_value = [
        {"name": "external_1", "type": "EXTERNAL_ACCESS"},
        {"name": "external_2", "type": "EXTERNAL_ACCESS"},
    ]
    mock_object_manager.return_value.describe.side_effect = ProgrammingError(
        "does not exist or not authorized"
    )
    ctx = mock_ctx()
    mock_connector.return_value = ctx

    with project_directory("snowpark_function_external_access") as project_dir:
        result = runner.invoke(
            [
                "snowpark",
                "deploy",
            ],
            catch_exceptions=False,
        )

    assert result.exit_code == 0, result.output
    assert ctx.get_queries() == [
        "create stage if not exists dev_deployment comment='deployments managed by snowcli'",
        f"put file://{Path(project_dir).resolve()}/app.zip @dev_deployment/my_snowpark_project"
        f" auto_compress=false parallel=4 overwrite=True",
        dedent(
            """\
            create or replace function func1(a string, b variant)
            returns string
            language python
            runtime_version=3.8
            imports=('@dev_deployment/my_snowpark_project/app.zip')
            handler='app.func1_handler'
            packages=()
            external_access_integrations=(external_1,external_2)
            secrets=('cred'=cred_name,'other'=other_name)
            """
        ).strip(),
    ]


@mock.patch("snowflake.connector.connect")
@mock.patch("snowflake.cli.plugins.snowpark.commands.ObjectManager")
def test_deploy_function_secrets_without_external_access(
    mock_object_manager,
    mock_conn,
    runner,
    mock_ctx,
    project_directory,
    snapshot,
):
    mock_object_manager.return_value.show.return_value = [
        {"name": "external_1", "type": "EXTERNAL_ACCESS"},
        {"name": "external_2", "type": "EXTERNAL_ACCESS"},
    ]
    ctx = mock_ctx()
    mock_conn.return_value = ctx

    with project_directory("snowpark_function_secrets_without_external_access"):
        result = runner.invoke(
            [
                "snowpark",
                "deploy",
            ],
        )

    assert result.exit_code == 1, result.output
    assert result.output == snapshot


@mock.patch("snowflake.connector.connect")
def test_deploy_function_no_changes(
    mock_connector,
    runner,
    mock_ctx,
    mock_cursor,
    project_directory,
):
    rows = [
        ("packages", '["foo=1.2.3", "bar>=3.0.0"]'),
        ("handler", "app.func1_handler"),
        ("returns", "string"),
    ]

    queries, result, project_dir = _deploy_function(
        rows,
        mock_connector,
        runner,
        mock_ctx,
        mock_cursor,
        project_directory,
        "--replace",
    )

    assert result.exit_code == 0, result.output
    assert json.loads(result.output) == [
        {
            "object": "func1(a string default 'default value', b variant)",
            "status": "packages updated",
            "type": "function",
        }
    ]
    assert queries == [
        "create stage if not exists dev_deployment comment='deployments managed by snowcli'",
        f"put file://{Path(project_dir).resolve()}/app.zip @dev_deployment/my_snowpark_project auto_compress=false parallel=4 overwrite=True",
    ]


@mock.patch("snowflake.connector.connect")
def test_deploy_function_needs_update_because_packages_changes(
    mock_connector,
    runner,
    mock_ctx,
    mock_cursor,
    project_directory,
):
    rows = [
        ("packages", '["foo=1.2.3"]'),
        ("handler", "main.py:app"),
        ("returns", "table(variant)"),
    ]

    queries, result, project_dir = _deploy_function(
        rows,
        mock_connector,
        runner,
        mock_ctx,
        mock_cursor,
        project_directory,
        "--replace",
    )

    assert result.exit_code == 0, result.output
    assert json.loads(result.output) == [
        {
            "object": "func1(a string default 'default value', b variant)",
            "status": "definition updated",
            "type": "function",
        }
    ]
    assert queries == [
        "create stage if not exists dev_deployment comment='deployments managed by snowcli'",
        f"put file://{Path(project_dir).resolve()}/app.zip @dev_deployment/my_snowpark_project auto_compress=false parallel=4 overwrite=True",
        dedent(
            """\
            create or replace function func1(a string default 'default value', b variant)
            returns string
            language python
            runtime_version=3.10
            imports=('@dev_deployment/my_snowpark_project/app.zip')
            handler='app.func1_handler'
            packages=('foo=1.2.3','bar>=3.0.0')
            """
        ).strip(),
    ]


@mock.patch("snowflake.connector.connect")
def test_deploy_function_needs_update_because_handler_changes(
    mock_connector,
    runner,
    mock_ctx,
    mock_cursor,
    project_directory,
):
    rows = [
        ("packages", '["foo=1.2.3", "bar>=3.0.0"]'),
        ("handler", "main.py:oldApp"),
        ("returns", "table(variant)"),
    ]

    queries, result, project_dir = _deploy_function(
        rows,
        mock_connector,
        runner,
        mock_ctx,
        mock_cursor,
        project_directory,
        "--replace",
    )

    assert result.exit_code == 0, result.output
    assert json.loads(result.output) == [
        {
            "object": "func1(a string default 'default value', b variant)",
            "status": "definition updated",
            "type": "function",
        }
    ]
    assert queries == [
        "create stage if not exists dev_deployment comment='deployments managed by snowcli'",
        f"put file://{Path(project_dir).resolve()}/app.zip @dev_deployment/my_snowpark_project"
        f" auto_compress=false parallel=4 overwrite=True",
        dedent(
            """\
            create or replace function func1(a string default 'default value', b variant)
            returns string
            language python
            runtime_version=3.10
            imports=('@dev_deployment/my_snowpark_project/app.zip')
            handler='app.func1_handler'
            packages=('foo=1.2.3','bar>=3.0.0')
            """
        ).strip(),
    ]


@mock.patch("snowflake.connector.connect")
def test_execute_function(mock_connector, runner, mock_ctx):
    ctx = mock_ctx()
    mock_connector.return_value = ctx
    result = runner.invoke(
        [
            "snowpark",
            "execute",
            "function",
            "functionName(42, 'string')",
        ]
    )

    assert result.exit_code == 0, result._output
    assert ctx.get_query() == "select functionName(42, 'string')"


def _deploy_function(
    rows,
    mock_connector,
    runner,
    mock_ctx,
    mock_cursor,
    project_directory,
    *args,
):
    ctx = mock_ctx(mock_cursor(rows=rows, columns=[]))
    mock_connector.return_value = ctx
    with mock.patch("snowflake.cli.plugins.snowpark.commands.ObjectManager") as om:
        om.return_value.describe.return_value = rows

        with project_directory("snowpark_functions") as temp_dir:
            (Path(temp_dir) / "requirements.snowflake.txt").write_text(
                "foo=1.2.3\nbar>=3.0.0"
            )
            result = runner.invoke(
                [
                    "snowpark",
                    "deploy",
                    "--format",
                    "json",
                    *args,
                ]
            )
    queries = ctx.get_queries()
    return queries, result, temp_dir
