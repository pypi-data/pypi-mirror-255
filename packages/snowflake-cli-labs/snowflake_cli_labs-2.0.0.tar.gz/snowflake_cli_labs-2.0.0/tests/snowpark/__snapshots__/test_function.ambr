# serializer version: 1
# name: test_deploy_function_secrets_without_external_access
  '''
  ╭─ Error ──────────────────────────────────────────────────────────────────────╮
  │ func1 defined with secrets but without external integration.                 │
  ╰──────────────────────────────────────────────────────────────────────────────╯
  
  '''
# ---
